packages/http-client-java/generator/http-client-generator-test/src/main/java/com/cadl/armresourceprovider/models/PriorityModel.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+
+package com.cadl.armresourceprovider.models;
+
+import com.azure.core.util.ExpandableEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Defines values for PriorityModel.
+ */
+public final class PriorityModel implements ExpandableEnum<Integer> {
+    private static final Map<Integer, PriorityModel> VALUES = new ConcurrentHashMap<>();
+
+    /**
+     * Static value 0 for PriorityModel.
+     */
+    public static final PriorityModel HIGH = fromValue(0);
+
+    /**
+     * Static value 1 for PriorityModel.
+     */
+    public static final PriorityModel LOW = fromValue(1);
+
+    private final Integer value;
+
+    private PriorityModel(Integer value) {
+        this.value = value;
+    }
+
+    /**
+     * Creates or finds a PriorityModel.
+     * 
+     * @param value a value to look for.
+     * @return the corresponding PriorityModel.
+     */
+    @JsonCreator
+    public static PriorityModel fromValue(Integer value) {
+        if (value == null) {
+            throw new IllegalArgumentException("value can't be null");
+        }
```

Comment:
Unsure whether we'd use `Objects.requireNotNull`
https://github.com/microsoft/typespec/blob/311497246ad90cfff5629615ec6afcd7adb06602/packages/http-client-java/generator/http-client-generator-test/src/main/java/com/routes/RoutesClientBuilder.java#L190

One risk is that service returns an explicit `null` (though no idea any backend really do this). It is fine to throw exception, till we really see such service.